src/contractManagers/index.ts

```diff
@@ -0,0 +1,146 @@
+import cpkFactoryAbi from '../abis/CpkFactoryAbi.json'
+import multiSendAbi from '../abis/MultiSendAbi.json'
+import safeAbiV111 from '../abis/SafeAbiV1-1-1.json'
+import safeAbiV120 from '../abis/SafeAbiV1-2-0.json'
+import { NormalizedNetworkConfigEntry } from '../config/networks'
+import EthLibAdapter, { Contract } from '../ethLibAdapters/EthLibAdapter'
+import { Address } from '../utils/basicTypes'
+import CommonContractManager from './CommonContractManager'
+import ContractV111Manager from './ContractV111Manager'
+import ContractV120Manager from './ContractV120Manager'
+
+export interface ContractManagerProps {
+  ethLibAdapter: EthLibAdapter
+  network: NormalizedNetworkConfigEntry
+  ownerAccount: Address | undefined
+  saltNonce: string
+  isSafeApp: boolean
+  isConnectedToSafe: boolean
+}
+
+class ContractManager {
+  #contractVersionManager?: CommonContractManager
```

Comment:
See above about the name. It could be confusing to have this `ContractManager` contain a `CommonContractManager`.

Instead, the `ContractManager` has a `ModuleManager` that knows how it should deal with the modules in the Safe.

Reply:
I see the point, we can use `ModuleManager`. However, what do you think it could be a good name for the class in case we have to handle a new contract method in the future that is not related with modules?

The idea here was to have the main `ContractManager` with a `CommonContractManager` handling the methods that all the contract versions have in common, or a subclass of `CommonContractManager` in case the current version has some differences.